OptimizelySDK/Bucketing/DecisionService.cs

```diff
@@ -253,5 +253,133 @@ public void SaveVariation(Experiment experiment, Variation variation, UserProfil
+            var rolloutLength = rollout.Experiments.Count;
+
+            // For all rules before the everyone else rule
+            for (int i=0; i < rolloutLength - 1; i++)
+            {
+                var rolloutExperiment = rollout.Experiments[i];
+
+                if (ExperimentUtils.IsUserInExperiment(ProjectConfig, rolloutExperiment, filteredAttributes))
+                {
+                    Logger.Log(LogLevel.DEBUG, $"Attempting to bucket user \"{userId}\" into rollout rule {rolloutExperiment.Key}");
+                    variation = Bucketer.Bucket(ProjectConfig, rolloutExperiment, userId, userId);
+
+                    if (variation == null)
+                    {
+                        Logger.Log(LogLevel.DEBUG, $"User \"{userId}\" was excluded due to traffic allocation. Checking 'Eveyrone Else' rule now.");
+                        break;
+                    }
+
+                    return variation;
+                }
+                else
+                {
+                    Logger.Log(LogLevel.DEBUG, $"User \"{userId}\" did not meet the audience conditions to be in rollout rule {rolloutExperiment.Key}");
+                }
+            }
+
+            // Buket the user into the last rule which is everyone else rule
+            var everyoneElseRule = rollout.Experiments[rolloutLength - 1];
+            variation = Bucketer.Bucket(ProjectConfig, everyoneElseRule, userId, userId);
+
+            if (variation == null)
+            {
+                Logger.Log(LogLevel.DEBUG, $"User \"{userId}\" was excluded from \"Everyone Else\" rule for feature flag \"{featureFlag.Key}\"");
+            }
+
+            return variation;
+        }
+
+        /// <summary>
+        /// Get the variation if the user is bucketed for one of the experiments on this feature flag.
+        /// </summary>
+        /// <param name = "featureFlag" >The feature flag the user wants to access.</param>
+        /// <param name = "userId" >User Identifier</param>
+        /// <param name = "filteredAttributes" >The user's attributes. This should be filtered to just attributes in the Datafile.</param>
+        /// <returns>null if the user is not bucketed into the rollout or if the feature flag was not attached to a rollout.
+        /// otherwise the Variation the user is bucketed into</returns>
+        public virtual Variation GetVariationForFeatureExperiment(FeatureFlag featureFlag, string userId, UserAttributes filteredAttributes)
+        {
+            if (featureFlag.ExperimentIds == null || featureFlag.ExperimentIds.Count == 0)
+            {
+                Logger.Log(LogLevel.INFO, $"The feature flag \"{featureFlag.Key}\" is not used in any experiments.");
+                return null;
+            }
+
+            foreach (var experimentId in featureFlag.ExperimentIds)
+            {
+                var experiment = ProjectConfig.GetExperimentFromId(experimentId);
+
+                if (string.IsNullOrEmpty(experiment.Key))
```

Comment:
log the message.
Please log the message, experiment is not in the datafile.
https://github.com/optimizely/ruby-sdk/blob/master/lib/optimizely/decision_service.rb#L168

Reply:
The message has already been logged inside GetExperimentFromId()